USER_GUIDE.md

```diff
@@ -1,5 +1,56 @@
 # `sqld` User Guide
 
+## Configuring gRPC to use TLS
+
+Configure the Common Name (CN) of the server:
+
+```console
+export SERVER_CN=localhost
+```
+
+Generate a password that protects the keys:
+
+```console
+export PASSWORD=$(echo $RANDOM | shasum | head -c 20; echo)
+```
+
+Generate Certificate Authority private key (`ca.key`) and trust certificate (`ca.crt`):
+
+```console
+openssl genrsa -passout pass:$PASSWORD -des3 -out ca.key 4096
+openssl req -passin pass:$PASSWORD -new -x509 -days 3650 -key ca.key -out ca.crt -subj "/CN=${SERVER_CN}"
+```
+
+Generate Server Private Key (`server.key`):
+
+```console
+openssl genrsa -passout pass:$PASSWORD -des3 -out server.key 4096
+```
+
+Generate Server Certificate Signing Request (`server.csr`):
+
+```console
+openssl req -passin pass:$PASSWORD -new -key server.key -out server.csr -subj "/CN=${SERVER_CN}" -config etc/ssl.cnf
+```
+
+Generate Server Certificate (`server.crt`):
+
+```console
+openssl x509 -req -passin pass:$PASSWORD -days 3650 -in server.csr -CA ca.crt -CAkey ca.key -set_serial 01 -out server.crt -extensions req_ext -extfile etc/ssl.cnf
+```
+
+Convert Server Certificate to `.pem` format (`server.pem`):
+
+```console
+openssl pkcs8 -topk8 -nocrypt -passin pass:$PASSWORD -in server.key -out server.pem
+```
+
+Start a `sqld` server with TLS enabled for gRPC:
+
+```console
+cargo run -- --grpc-listen-addr 127.0.0.1:5001 --grpc-tls --grpc-cert-file server.pem --grpc-key-file server.key
```

Comment:
Perhaps worth transforming this guide into a script that a user can run to generate the key/cert?